### PR TITLE
FindScheduleCommand

### LINKED ISSUE #51: as a user, I can search for a staff's schedule on a particular day or date

### PR DESCRIPTION
**Pull request by mweeruien**
Completed implementation of FindScheduleCommand, as detailed in the modified UserGuide.

Closes #51

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -290,30 +273,41 @@ Examples:
...
+`findSchedule -d [DAY]-[SHIFT NUMBER]`
+`findSchedule -t [DAY]-[HH:mm]` (Note that this is in 24-hour format)
+
+Note that [DAY] refers to the day of the week, and it is case-insensitive. However, it should be spelt in full (e.g. MONDAY instead of Mon). 
+
+Examples:\
+`findSchedule -d monday-1` \
+`findSchedule -d TUESDAY-0` \
+`findSchedule -t wednesday-12:00`\
+`findSchedule -t THURSDAY-16:30`
```
Comment (fullfatwasabi) [inline docs/UserGuide.md:310]: Don't forget to add this to the summary at the bottom! :)

File: src/main/java/seedu/address/logic/commands/FindScheduleCommand.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+/**
+ * Class representing the view schedule command
+ * which views the schedule by Person.
+ */
+public class FindScheduleCommand extends Command {
+    public static final String DEFAULT_MESSAGE = "Staff working at that shift: \n";
+    public static final String COMMAND_WORD = "findSchedule";
+    public static final String HELP_MESSAGE = COMMAND_WORD + ": find the staff working at the specified shift\n\n"
+            + "Parameters:\n"
+            + PREFIX_DASH_DAY_SHIFT + " [DAY]-[SLOT NUMBER]\n"
```
Comment (fullfatwasabi) [inline src/main/java/seedu/address/logic/commands/FindScheduleCommand.java:26]: Hmm sorry not sure about the square brackets cause they're not optional! HAHA sorry I didn't see it just now 😭

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 3]
```diff
@@ -196,6 +198,44 @@ public static String parseDayOfWeek(String shiftDay) throws ParseException {
...
+        case "monday":
+        case "tuesday":
+        case "wednesday":
+        case "thursday":
+        case "friday":
+        case "saturday":
+        case "sunday":
+            break;
+        default: throw new ParseException(messageConstraints);
+        }
```
Comment (fullfatwasabi) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:228]: Hmm I think this is a weird use of a switch block. Maybe you could just use a normal check?
```suggestion
        List<String> strDays = Arrays.asList("Monday", "Tuesday", "Wednesday", "Thusday", "Friday", "Saturday", "Sunday" );
        if (!strDays.contains(strings[0])) {
            throw new ParseException(messageConstraints);
        }
```

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 4]
```diff
@@ -196,6 +198,44 @@ public static String parseDayOfWeek(String shiftDay) throws ParseException {
         return trimmedStr;
     }
 
+    /**
+     * Parses a {@code String dayOfWeek} into an {@code DayOfWeek}.
```
Comment (fullfatwasabi) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:202]: This doesnt actl return a DayOfWeek

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 3 above)
Comment (fullfatwasabi) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java:228]: feel free to use isValidDayOfWeek() in Shift.java too maybe might be neater actl
```suggestion
        if (!isValidDayOfWeek(strings[0]) {
            throw new ParseException(messageConstraints);
        }
```

Comment (Gabau) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java:228]: > Hmm I think this is a weird use of a switch block. Maybe you could just use a normal check?

If you are using the Arrays method make sure to keep to lower case. I think it is better to use isValidDayOfWeek. Keeps it to single point of failure. Using the isValidDayOfWeek means we do not need to convert to lower case, though it is probably better to do so.

**Pull request merged**